src/main/scala/com/rbmhtechnology/calliope/serializer/kafka/PayloadFormatSerializer.scala

```diff
@@ -48,22 +51,46 @@ class PayloadFormatSerializer[A <: AnyRef] private(serializer: PayloadSerializer
 }
 
 object PayloadFormatDeserializer {
-  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
-    new PayloadFormatDeserializer[A](DelegatingStringManifestPayloadSerializer(system), payloadMapper)
 
   def apply()(implicit system: ActorSystem): PayloadFormatDeserializer[AnyRef] =
     apply[AnyRef](identity)
 
-  def create[A](payloadMapper: JFunction[AnyRef, A], system: ActorSystem): PayloadFormatDeserializer[A] =
-    apply(payloadMapper.apply)(system)
+  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
+    instance(payloadMapper, err => throw err)
+
+  def option[A](payloadMapper: PartialFunction[AnyRef, A])(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
+    apply(payloadMapper.lift)
+
+  def filterError[A](payloadMapper: PartialFunction[AnyRef, A], onError: Throwable => Unit)(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
```

Comment:
On second thought I have n issue in how to define the `attempt` which should return a try and does the payload mapping with a partial function. 

It would probably look like so, wouldn't it:
`attemptFilterError[A](payloadMapper: PartialFunction[AnyRef, A]): PayloadFormatDeserializer[Try[Option[A]]]`

This looks a bit of, a bit simpler could be:
`attemptFilterError[A](payloadMapper: PartialFunction[AnyRef, A]): PayloadFormatDeserializer[Try[A]]`
but in this case anything not handled by the payloadMapper would then result in a `Failure` perhaps with an `IllegalArgumentException`?